### DISCUSSION TITLE
Unit Testing w/ Data Wrangler

### DISCUSSION THREAD

**Discussion by bryner84** [aws/aws-sdk-pandas#2613] [Q&A]
Hello, I'm looking for some input on how I might write some unit tests with this library.  Specifically, with database interactions.  I'm using the following `wr.postgresql.connect(DB_CONNECTION_NAME)` which requires Glue.  I thought about using a secret as it the other option to connect.  If I did that, would I just mock the secret?  Ultimately, I just want to test some reads/writes to a local PG instance.

Any insights would be appreciated.

**Comment by AdrianoNicolucci** [comment]
What is the purpose of these specific unit tests? Are you trying to test to make sure you can connect to your database? I would classify this more of a integration test rather than unit test.  Are you doing any transformations? I think it would be more important to develop unit tests to make sure any business logic you are developing in python. The developers of the library have already built unit tests for the functions themselves so I'm not sure what a mock connection would really prove?
